### PR TITLE
Allow filtering the questions API by date range

### LINKED ISSUE #3332: Allow filtering the questions API by date range
The `created` and `updated` datetime fields should be filterable by ranges or other date-appropriate Django filter types (e.g. `month`, `hour`, `year`, etc.).

### PR DESCRIPTION
**Pull request by vaishnavipamulapati**
Fixes #3332

cc @pmac

### CODE REVIEW

**Comment by vaishnavipamulapati** [comment]
- Is it okay to assume that the range input will be a python list of `datetime` objects, or should I parse the dates from strings?
- Should this put to use somewhere in specific on the frontend?

File: kitsune/questions/api.py [hunk 1]
```diff
@@ -224,6 +228,14 @@ def filter_metadata(self, queryset, value):
 
         return queryset
 
+    def filter_created_between(self, queryset, range):
+        questions_created_between = Question.objects.filter(created__range=range)
```
Comment (safwanrahman) [inline kitsune/questions/api.py]: do not use built in variable `range`

Comment (vaishnavipamulapati) [reply kitsune/questions/api.py]: thanks!

**Comment by akatsoulas** [comment]
Hi all. The API now supports `lt` and `gt` filters in the queryset for both `created` and `updated` fields.
The query can have the form of `YYYY-MM-DD HH:MM:SS` and the time part can be optional. A fully formulated query can look like `created__lt=YYYY-MM-DD HH:MM:SS`. By combining both `lt` and `gt` filters you can get a date range filtering. Because of this I am closing this PR. Thank you @vaishnavipamulapati for submitting this patch!

**Comment by akatsoulas** [comment]
The filtering works for both Question and Answers.

**Pull request closed without merging**